compose/snippets/src/main/java/com/example/compose/snippets/components/Menu.kt

```diff
@@ -0,0 +1,157 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.compose.snippets.components
+
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.ArrowDropDown
+import androidx.compose.material.icons.filled.MoreVert
+import androidx.compose.material3.Button
+import androidx.compose.material3.DropdownMenu
+import androidx.compose.material3.DropdownMenuItem
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.material3.FloatingActionButton
+import androidx.compose.material3.Icon
+import androidx.compose.material3.IconButton
+import androidx.compose.material3.Text
+import androidx.compose.material3.TextField
+import androidx.compose.material3.TopAppBar
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+
+@Composable
+fun MenusExamples() {
+    var currentExample by remember { mutableStateOf<(@Composable () -> Unit)?>(null) }
+
+    // Display the current example and the button to close it.
+    Box(modifier = Modifier.fillMaxSize()) {
+        currentExample?.let {
+            it()
+            FloatingActionButton(
+                onClick = { currentExample = null },
+                modifier = Modifier
+                    .align(Alignment.BottomEnd)
+                    .padding(16.dp)
+            ) {
+                Text(text = "Close example", modifier = Modifier.padding(16.dp))
+            }
+            return
+        }
+    }
+
+    // Display the list of available examples.
+    Column(modifier = Modifier.padding(16.dp)) {
+        Button(onClick = { currentExample = { BasicDropdownMenu() } }) {
+            Text("Action Bar Menu")
+        }
+        Button(onClick = { currentExample = { TextFieldDropdownMenu() } }) {
+            Text("Options Menu")
+        }
+    }
+}
+
+@OptIn(ExperimentalMaterial3Api::class)
+// [START android_compose_components_basicdropdownmenu]
+@Composable
+fun BasicDropdownMenu() {
+    var shouldDisplayMenu by remember { mutableStateOf(false) }
+
+    TopAppBar(
+        title = { Text(text = "Basic menu example") },
+        actions = {
+            IconButton(onClick = { shouldDisplayMenu = !shouldDisplayMenu }) {
```

Comment:
I'd recommend adding a second action, thereby making it clearer that the IconButton+DropdownMenu should be wrapped by a Box to correctly set the anchor for the DropdownMenu:

```
        actions = {
            IconButton(onClick = { shouldDisplayMenu = !shouldDisplayMenu }) {
                Icon(Icons.Default.StarOutline, stringResource( R.string.cd_star))
            }

            Box {
                IconButton(onClick = { shouldDisplayMenu = !shouldDisplayMenu }) {
                    Icon(Icons.Default.MoreVert, stringResource(R.string.cd_more_options))
                }
                DropdownMenu(
                    expanded = shouldDisplayMenu,
                    onDismissRequest = { shouldDisplayMenu = false }
                ) {
                  DropdownMenuItem(
                      text = { Text("Refresh") },
                      onClick = { /* Handle refresh! */ }
                  )
                  DropdownMenuItem(
                      text = { Text("Settings") },
                      onClick = { /* Handle settings! */ }
                  )
                }
            }

```